### PR TITLE
feat(API): Implement Linframe type

### PR DESCRIPTION
**Pull request by bcornish**
- [x] This contribution adheres to [CONTRIBUTING.md](https://github.com/ni/nixnet-python/blob/master/CONTRIBUTING.rst).
- [ ] New tests have been created for any new features or regression tests for bugfixes.
- [x] `tox` successfully runs, including unit tests and style checks (see [CONTRIBUTING.md](https://github.com/ni/nixnet-python/blob/master/CONTRIBUTING.rst)).

### What does this Pull Request accomplish?

Adds/ updates LinFrame

### Why should this Pull Request be merged?

Adds LinFrame support.

### What testing has been done?

Doc tests

### CODE REVIEW

**Comment by epage** [comment]
This was forked off of #142

File: nixnet/types.py [hunk 1]
```diff
@@ -495,6 +496,130 @@ def __repr__(self):
...
+
+    @classmethod
+    def from_raw(cls, frame):
+        # type: (RawFrame) -> LinFrame
+        """Convert from RawFrame.
+
+        >>> raw = RawFrame(5, 2, constants.FrameType.LIN_DATA, _cconsts.NX_FRAME_FLAGS_TRANSMIT_ECHO, 0, b'\x01')
+        >>> LinFrame.from_raw(raw)
+        LinFrame(identifier=2, echo=True, timestamp=0x5, len(payload)=1)
+        """  # NOQA: E501
```
Comment (epage) [inline nixnet/types.py]: Why was this needed?

File: nixnet/types.py [hunk 2]
```diff
@@ -495,6 +496,130 @@ def __repr__(self):
             self.rx_err_count)
 
 
+class LinFrame(object):
```
Comment (epage) [inline nixnet/types.py]: When done, could you squash your commits?

File: nixnet/types.py [hunk 3]
```diff
@@ -495,6 +496,130 @@ def __repr__(self):
...
+        >>> LinFrame.from_raw(raw)
+        LinFrame(identifier=2, echo=True, timestamp=0x5, len(payload)=1)
+        """  # NOQA: E501
+        identifier = frame.identifier & cls._FRAME_ID_MASK
+        lin_frame = LinFrame(identifier, constants.FrameType(frame.type), frame.payload)
+        lin_frame.timestamp = frame.timestamp
+        lin_frame.echo = bool(frame.flags & _cconsts.NX_FRAME_FLAGS_TRANSMIT_ECHO)
+        lin_frame.eventslot = bool(frame.flags & _cconsts.NX_FRAME_FLAGS_LIN_EVENT_SLOT)
+        if lin_frame.eventslot:
+            lin_frame.eventid = frame.info
```
Comment (epage) [inline nixnet/types.py]: This is missing coverage
https://coveralls.io/builds/12752500/source?filename=nixnet%2Ftypes.py#L550

File: nixnet/types.py [hunk 4]
```diff
@@ -495,6 +496,130 @@ def __repr__(self):
...
+        else:
+            lin_frame.eventid = 0
+
+        return lin_frame
+
+    def to_raw(self):
```
Comment (epage) [inline nixnet/types.py]: A couple parts of `to_raw` are missing coverage
https://coveralls.io/builds/12752500/source?filename=nixnet%2Ftypes.py#L556

File: nixnet/types.py [hunk 5]
```diff
@@ -495,6 +496,130 @@ def __repr__(self):
...
+                self.echo == other_frame.echo,
+                self.type == other_frame.type,
+                self.timestamp == other_frame.timestamp,
+                self.eventslot == other_frame.eventslot,
+                self.eventid == other_frame.eventid,
+                self.payload == other_frame.payload))
+        else:
+            return NotImplemented
+
+    def __repr__(self):
```
Comment (epage) [inline nixnet/types.py]: A couple parts of your repr are missing coverage
https://coveralls.io/builds/12752500/source?filename=nixnet%2Ftypes.py#L592

File: nixnet/types.py (hunk 1 above)
Comment (epage) [inline nixnet/types.py]: (The NOQA)

File: nixnet/types.py (hunk 4 above)
Comment (bcornish) [inline nixnet/types.py]: I'm not sure how to add test coverage for
`_errors.check_for_error(_cconsts.NX_ERR_UNDEFINED_FRAME_ID)`

I hand-tested it to confirm the error is thrown.

Comment (epage) [inline nixnet/types.py]: There should be similar tests for CanFrame, try test_can_identifier_extended_overflow

Comment (bcornish) [inline nixnet/types.py]: Added in my latest commit.
